File: jobscheduling.cpp
//============================================================================
// Name        : DAA.cpp
// Author      : ajinkya
// Version     :
// Copyright   : Your copyright notice
// Description : Hello World in C++, Ansi-style
//============================================================================
// Job Scheduling Algorithm

#include <iostream>
#include <algorithm>
using namespace std;
//represent a job
struct Job
{
	char id;
	int dead;
	int profit;
};
//function for comparison for sorting
bool comparison(Job a, Job b){
return (a.profit>b.profit);
}
void JobScheduling(Job arr[], int n){
	sort(arr, arr+n ,comparison );
	int result[n];
	bool slot[n];

	for (int i = 0; i < n; ++i) {
		slot[i]=false;
	}
	for (int i = 0; i < n; ++i) {
		for(int j=min(n,arr[i].dead)-1;j>=0;j--)
		{
			if(slot[j]==false)
			{
				result[j]=i;
				slot[j]=true;
				break;
			}
		}
	}
	 // Print the result
	    for (int i=0; i<n; i++)
	       if (slot[i])
	         cout << arr[result[i]].id << " ";
}
int main() {
	Job	 arr[] = { {'a', 2, 100}, {'b', 1, 19}, {'c', 2, 27},
            {'d', 1, 25}, {'e', 3, 15}};
    int n = sizeof(arr)/sizeof(arr[0]);
    cout << "Following is maximum profit sequence of jobs";
    JobScheduling(arr, n);


	cout << "!!!Hello World!!!" << endl; // prints !!!Hello World!!!
	return 0;
}

File: JobScheduling.py
# -*- coding: utf-8 -*-
"""
Created on Sun Jan 27 12:20:22 2019

@author: HP
"""
def JobScheduling(arr , t):
    n=len(arr)
    
    for i in range(n):
        for j in range (n-i-1):
            if arr[j][2]< arr[j+1][2]:
                arr[j],arr[j+1]= arr[j+1],arr[j]
                
                
    result = [False]* t
    job = ['-1' ]*t

    for i in range(len(arr)):
        for j in range(min(t-1,arr[i][1]-1),-1,-1):
            if result[j] is False:
                result[j] = True
                job[j]=arr[i][0]
                break
     
        
    print (job)       
                
arr = [ ['a',2,100],
          ['b',1,19],
          ['c',2,27],
          ['d',1,25],
          ['e',3,15] ]

print("following is maximum profit sequence jobs")
JobScheduling(arr,3)